docs/source/developers/guide/introduction.rst

```diff
@@ -0,0 +1,142 @@
+.. Add an introduction to the project, why to get involved
+.. and a quick checklist of the contributing process.
+.. Add the Table of Contents. All detailed explanations
+.. should be added as a link in the content.
+
+
+.. _introduction:
+
+***********************
+New Contributor's Guide
+***********************
+
+This guide is meant to be a resource for contributing to
+`Arrow <https://arrow.apache.org//>`_ for new contributors.
+
+Starting to contribute to a project like Apache Arrow can look
+scary. Taking small steps will help making this tack easier.
+
+
+Why contribute to Arrow?
+========================
+There can be various reasons why someone would want to contribute
+to Arrow:
+
+* You find the project interesting and would like to try making
+  a contribution to learn more about the library and grow your skills.
+
+* You use Arrow in the project you are working on and you would like
+  to make a new feature or fix a bug yourself.
+
+Read more about the project in the :ref:`about_arrow` section.
+
+Quick Reference
+===============
+
+Here are the basic steps needed to get set up and contribute to Arrow.
+This is meant as a checklist and also to have an overall picture.
+
+For complete instructions please follow :ref:`step_by_step` (a
+step-by-step guide) or R and Python :ref:`tutorial-index` for an example
+of adding a basic feature.
+
+
+#. **Install and set up Git, fork the Arrow repository**
+
+   See detailed instructions on how to :ref:`set-up` Git and fork the
+   Arrow repository.
+
+#. **Build Arrow**
+
+   This can be quite easy or a real blocker. Be prepared to dig through the
+   additional flags for cmake in the C++ build guide and read the doc carefully
+   to not miss any suggestions already given there. Anytime you are stuck,
+   reach out to us via appropriate :ref:`communication` channel.
+
+   See detailed instructions on how to :ref:`build-arrow`.
+ 
+#. **Run the tests**
+   
+   Run the tests from a terminal for Python
+
+   .. code-block::
+
+      $ pytest pyarrow
+
+   or R
+
+   .. code-block::
+
+      $ devtools::test()
+
+#. **Find an issue, create a new branch and work on the problem**
+
+   **Finding an issue**
+
+   Not so easy, we agree.
+   Read through the :ref:`finding-issues` section to get some ideas.
+
+   **Finding your way through the project**
+
+   The first step when starting a new project is the hardest. We wrote some
+   help guides that we used when we were looking for solutions and we hope
+   they will help.
+
+   Read through :ref:`solving` section.
+
+   **Communication**
+
+   Communication is very important. You may need some help solving the problem
+   (happening to developers all the time). Also, when you have an issue you want
+   to solve it is advisable to let the team know you are working on it and may
+   need some help.
+
+   See possible channels of :ref:`communication`.
+
+#. **Once you found the solution, write and run the test**
```

Comment:
This implies there is a single solution (and that one exists!) and a single test.

I would rephrase it for example as:
> Once you implemented the planned fix or feature, write and run tests for it